### PR TITLE
Migrate to nvm

### LINKED ISSUE #192: Switch to nvm for Node installation

### PR DESCRIPTION
**Pull request by dimkarakostas**
Fix #192

### CODE REVIEW

**Comment by dimkarakostas** [comment]
Fixes #192

File: rupture [hunk 1]
```diff
@@ -42,9 +42,21 @@ where <parameter> is:
           in relative directories."
 }
 
+nvm_setup() {
+    curl https://raw.githubusercontent.com/creationix/nvm/v0.11.1/install.sh | bash
```
Comment (dionyziz) [inline rupture]: Use the [latest version 0.34.4](https://github.com/creationix/nvm/blob/master/README.markdown)

File: realtime/setup.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+BASEDIR=$(dirname -- "$(readlink -f -- "${BASH_SOURCE}")")
+
+echo '[-] Setting up realtime...'
+
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
```
Comment (dionyziz) [inline realtime/setup.sh]: Can we DRY this somehow?

Comment (dimkarakostas) [inline realtime/setup.sh]: I don't see how. The problem is that during installation nvm exports the necessary env vars in the profile of the user's shell. I don't know how we can access that inside of the shell script without knowing which shell the user uses by default. This solution is ugly but it's the only one I have found that works consistently across systems.

**Comment by esarafianou** [comment]
LGTM

File: install.sh [hunk 3]
```diff
@@ -20,6 +13,17 @@ function activate_virtualenv {
     pip install -r requirements.txt
 }
 
+install_nvm_node() {
+    sudo apt-get install -y build-essential libssl-dev curl
+    curl https://raw.githubusercontent.com/creationix/nvm/v0.33.2/install.sh | bash
```
Comment (dionyziz) [inline install.sh]: Check if `$NVM_DIR` is already defined and working before installing nvm.

File: install.sh [hunk 4]
```diff
@@ -66,53 +74,45 @@ fi
...
-        sudo gem install syslog-logger #install syslog-logger from local directory or remote server
-        sudo apt-get install build-essential ruby-dev libpcap-dev #install dependencies
-        sudo gem install bettercap
+        npm install)
+        install_bettercap
         (cd realtime
-        npm install) ##install required packages for compilation of the client code
-	 install_python
+        npm install)
+        install_python
```
Comment (dionyziz) [inline install.sh]: DRY

**Comment by dionyziz** [comment]
LGTM

**Pull request merged**